File: Linked_Lists/go/Stack.go
package types

import (
  "fmt"
)


// Stack implementation
type Stack struct {
  elements []int
  top int
  size int
}


func (s *Stack) Push(x int) {
  s.elements = append(s.elements, x)
  s.top = x
  s.size++
}


func (s *Stack) Pop() (int, bool) {
  top := s.top

  if s.IsEmpty() {
    return 0, false
  }

  s.elements = s.elements[:s.size-1]
  s.size--

  if s.IsEmpty() {
    return top, true
  }
  s.top = s.elements[s.size-1]

  return top, true
}


func (s *Stack) IsEmpty() bool {
  return s.size == 0
}


func PrintStack(s Stack) {
  fmt.Println(s.elements)
}

File: Linked_Lists/python/Stack.py
class Stack(object):
    def __init__(self):
        self.top = None
        self.elements = []

    def push(self, x):
        self.elements.append(x)
        self.top = x

    def pop(self):
        top = self.top
        length = len(self.elements)

        if not length:
            return None

        self.elements = self.elements[:length-1]

        if length == 1:
            return top

        self.top = self.elements[length-2]

        return top
